pkg/cmdutil/validators/synonyms.go

```diff
@@ -0,0 +1,133 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+)
+
+type SaveOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	SearchClient func() (*search.Client, error)
+
+	Indice              string
+	SynonymID           string
+	ForwardToReplicas   bool
+	SynonymType         SynonymType
+	SynonymInput        string
+	SynonymWord         string
+	SynonymPlaceholder  string
+	Synonyms            []string
+	SynonymCorrections  []string
+	SynonymReplacements []string
+	Synonym             search.Synonym
+}
+
+// Defining new type that implements pflag.Value interface with String, Set and Type
+// https://stackoverflow.com/questions/50824554/permitted-flag-values-for-cobra
+type SynonymType string
+
+const (
+	// Matching API https://www.algolia.com/doc/api-reference/api-methods/save-synonym/#method-param-type
+	Regular        string = "synonym"
+	OneWay         string = "oneWaySynonym"
+	AltCorrection1 string = "altCorrection1"
+	AltCorrection2 string = "altCorrection2"
+	Placeholder    string = "placeholder"
+)
+
+func (e *SynonymType) String() string {
+	return string(*e)
+}
+
+func (e *SynonymType) Set(v string) error {
+	if v == "" {
+		*e = SynonymType(v)
+		return nil
+	}
+
+	switch v {
+	case Regular, OneWay, AltCorrection1, AltCorrection2, Placeholder:
+		*e = SynonymType(v)
+		return nil
+	default:
+		return fmt.Errorf(`must be one of "regular", "one-way", "alt-correction1", "alt-correction2" or "placeholder"`)
+	}
+}
+
+func (e *SynonymType) Type() string {
+	return "SynonymType"
+}
+
+func ValidateFlags(options SaveOptions) (search.Synonym, error) {
```

Comment:
I'd love to split that with a validation function and a facade that'll create the synonym. I found this that can handle complex validation (with conditionals) https://github.com/go-playground/validator but it may bring more complexity to the code 🤔